### DISCUSSION TITLE
@bablel/plugin-transform-modules-commonjs

### DISCUSSION THREAD

**Discussion by gokulk04** [babel/babel#12969] [Q&A]
Hi! I'm currently working on a monorepo where we have our web app compiled in esnext, with the rest of the codebase compiled in commonjs. We needed to use the `@bablel/plugin-transform-modules-commonjs` plugin to allow our imports to work properly unless I change the web app to also compile in commonjs. Otherwise, I would end up with an `Uncaught ReferenceError: exports is not defined at` error. Is there a better way to address this without the plugin? The plugin bloats our webpack bundle by a good amount, causing our app to run slower. Thanks!

**Comment by nicolo-ribaudo** [comment]
I think that the problem is that Babel injects ESM code in files that should have been CJS. Can you try adding `sourceType: "unambiguous"` to your Babel config?

**Reply by gokulk04** [reply]
Thanks for the response :) I'll give this a shot. I think the root of the issue for our webpack bundle being so large is becuase commonjs doesnt allow code-splitting. So even though our web app is being compiled in `esnext` as you can see by our tsconfig below, adding the `@bablel/plugin-transform-modules-commonjs` in our config-overrides.js transforms it to commonjs, so dynamic imports no longer get enabled. 

```{
  "compilerOptions": {
    "baseUrl": "src",
    "target": "es6",
    "lib": ["dom", "es6"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": false,
    "forceConsistentCasingInFileNames": false,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "composite": false,
    "noEmit": true,
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": false
  },
  "references": [
    {
      "path": "../../datamodel"
    }
  ],
  "include": ["./src"]
}
```
I'm wondering if adding the `babel-plugin-dynamic-import-node` module to our config would help
